### PR TITLE
clean TODOs and implement back some Bootloader API calls

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

File: src/lib/bootloader/auto_client.rb [hunk 1]
```diff
@@ -26,19 +26,10 @@ def run
...
-      proposed_configuration = BootloaderFactory.bootloader_by_name(imported_configuration.name)
-      proposed_configuration.propose
-
-      proposed_configuration.merge(imported_configuration)
-      BootloaderFactory.current = proposed_configuration
+      Yast::Bootloader.Import(data)
 
       Yast::PackagesProposal.AddResolvables("yast2-bootloader",
-        :package, proposed_configuration.packages)
+        :package, BootloaderFactory.current..packages)
```
Comment (imobach) [inline src/lib/bootloader/auto_client.rb]: It looks like you have "two points" instead of only one.

Comment (jreidinger) [inline src/lib/bootloader/auto_client.rb]: ah, you are right, typo

File: src/lib/bootloader/grub2base.rb [hunk 2]
```diff
@@ -153,17 +153,21 @@ def merge_password(other)
...
 
       log.info "before merge default #{default.inspect}"
       log.info "before merge other #{other.inspect}"
 
-      # TODO: other kernel flavors probably will be also needed
-      unless other.kernel_params.serialize.empty?
-        new_kernel_params = default.kernel_params.serialize + " " + other.kernel_params.serialize
-        default.kernel_params.replace(new_kernel_params)
+      KERNEL_FLAVORS_METHODS.each do |method|
+        next if  other.public_send(method).empty?
```
Comment (imobach) [inline src/lib/bootloader/grub2base.rb]: np: you have two spaces instead of just one :)

File: src/modules/Bootloader.rb [hunk 3]
```diff
@@ -201,9 +168,16 @@ def Read
     def Reset
       return if Mode.autoinst
       log.info "Reseting configuration"
```
Comment (imobach) [inline src/modules/Bootloader.rb]: Rese*tt*ing

**Comment by imobach** [comment]
If Travis is ok, LGTM.

File: src/lib/bootloader/grub2base.rb (hunk 2 above)
Comment (jreidinger) [inline src/lib/bootloader/grub2base.rb]: good catch ;)

**Pull request merged**